torchvision/datasets/standford_cars.py

```diff
@@ -0,0 +1,145 @@
+            downloaded again."""
+
+    urls = (
+        "https://ai.stanford.edu/~jkrause/car196/cars_test.tgz",
+        "https://ai.stanford.edu/~jkrause/car196/cars_train.tgz",
+    )  # test and train image urls
+
+    md5s = (
+        "4ce7ebf6a94d07f1952d94dd34c4d501",
+        "065e5b463ae28d29e77c1b4b166cfe61",
+    )  # md5checksum for test and train data
+
+    annot_urls = (
+        "https://ai.stanford.edu/~jkrause/car196/cars_test_annos_withlabels.mat",
+        "https://ai.stanford.edu/~jkrause/cars/car_devkit.tgz",
+    )  # annotations and labels for test and train
+
+    annot_md5s = (
+        "b0a2b23655a3edd16d84508592a98d10",
+        "c3b158d763b6e2245038c8ad08e45376",
+    )  # md5 checksum for annotations
+
+    def __init__(
+        self,
+        root: str,
+        train: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ) -> None:
+
+        try:
+            from scipy.io import loadmat
+
+            self._loadmat = loadmat
+        except ImportError:
+            raise RuntimeError("Scipy is not found. This dataset needs to have scipy installed: pip install scipy")
+
+        super().__init__(root, transform=transform, target_transform=target_transform)
+
+        self.train = train
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        self._samples = self._make_dataset()
+        self.classes = self._get_classes_name()  # class_id to class_name mapping
+
+    def _get_class_names(self) -> dict:
+        """
+        Returns Mapping of class ids to class names in form of Dictionary
+        """
+        meta_data = self._loadmat(os.path.join(self.root, "devkit/cars_meta.mat"))
+        class_names = meta_data["class_names"][0]
+        return {class_name[0].replace(" ", "_").replace("/", "_"): i for i, class_name in enumerate(class_names)}
+
+    def _make_dataset(self):
+        """
+        Returns Annotations for training data and testing data
+        """
+        annotations = None
+        if self.train:
+            annotations = self._loadmat(os.path.join(self.root, "devkit/cars_train_annos.mat"))
+        else:
+            annotations = self._loadmat(os.path.join(self.root, "cars_test_annos_withlabels.mat"))
+        samples = []
+        annotations = annotations["annotations"][0]
+        for index in range(len(annotations)):
+            target = annotations[index][4][0, 0]
+            image_file = annotations[index][5][0]
+            # Beware: Stanford cars targets starts at 1
+            target = target - 1
+            samples.append((image_file, target))
+        return samples
+
+    def __len__(self) -> int:
+        return len(self._samples)
+
+    def __getitem__(self, idx: int) -> (Image, int):
```

Comment:
We can't know the output type here due to `transform` and `target_transform`.

```suggestion
    def __getitem__(self, idx: int) -> Tuple[Any, Any]:
```